ml-algorithms/src/test/java/org/opensearch/ml/engine/httpclient/MLHttpClientFactoryTests.java

```diff
@@ -29,84 +35,115 @@ public void test_getSdkAsyncHttpClient_success() {
 
     @Test
     public void test_validateIp_validIp_noException() throws Exception {
-        AtomicBoolean privateIpEnabled = new AtomicBoolean(false);
-        MLHttpClientFactory.validate("http", "api.openai.com", 80, privateIpEnabled);
+        MLHttpClientFactory.validate(HTTP, TEST_HOST, 80, PRIVATE_IP_DISABLED);
+        MLHttpClientFactory.validate(HTTP, "127.0.0.1", 80, PRIVATE_IP_ENABLED);
+        MLHttpClientFactory.validate(HTTP, "177.16.0.1", 80, PRIVATE_IP_DISABLED);
+        MLHttpClientFactory.validate(HTTP, "177.0.1.1", 80, PRIVATE_IP_DISABLED);
+        MLHttpClientFactory.validate(HTTP, "177.0.0.2", 80, PRIVATE_IP_DISABLED);
+        MLHttpClientFactory.validate(HTTP, "::ffff", 80, PRIVATE_IP_DISABLED);
     }
 
     @Test
     public void test_validateIp_rarePrivateIp_throwException() throws Exception {
-        AtomicBoolean privateIpEnabled = new AtomicBoolean(false);
         try {
-            MLHttpClientFactory.validate("http", "0254.020.00.01", 80, privateIpEnabled);
+            MLHttpClientFactory.validate(HTTP, "0254.020.00.01", 80, PRIVATE_IP_DISABLED);
         } catch (IllegalArgumentException e) {
             assertNotNull(e);
         }
 
         try {
-            MLHttpClientFactory.validate("http", "172.1048577", 80, privateIpEnabled);
+            MLHttpClientFactory.validate(HTTP, "172.1048577", 80, PRIVATE_IP_DISABLED);
         } catch (Exception e) {
             assertNotNull(e);
         }
 
         try {
-            MLHttpClientFactory.validate("http", "2886729729", 80, privateIpEnabled);
+            MLHttpClientFactory.validate(HTTP, "2886729729", 80, PRIVATE_IP_DISABLED);
+        } catch (IllegalArgumentException e) {
+            assertNotNull(e);
+        }
+
+        try {
+            MLHttpClientFactory.validate(HTTP, "192.11010049", 80, PRIVATE_IP_DISABLED);
+        } catch (IllegalArgumentException e) {
+            assertNotNull(e);
+        }
+
+        try {
+            MLHttpClientFactory.validate(HTTP, "3232300545", 80, PRIVATE_IP_DISABLED);
         } catch (IllegalArgumentException e) {
             assertNotNull(e);
         }
 
         try {
-            MLHttpClientFactory.validate("http", "192.11010049", 80, privateIpEnabled);
+            MLHttpClientFactory.validate(HTTP, "0:0:0:0:0:ffff:127.0.0.1", 80, PRIVATE_IP_DISABLED);
         } catch (IllegalArgumentException e) {
             assertNotNull(e);
         }
 
         try {
-            MLHttpClientFactory.validate("http", "3232300545", 80, privateIpEnabled);
+            MLHttpClientFactory.validate(HTTP, "153.24.76.232", 80, PRIVATE_IP_DISABLED);
         } catch (IllegalArgumentException e) {
             assertNotNull(e);
         }
 
         try {
-            MLHttpClientFactory.validate("http", "0:0:0:0:0:ffff:127.0.0.1", 80, privateIpEnabled);
+            MLHttpClientFactory.validate(HTTP, "177.0.0.1", 80, PRIVATE_IP_DISABLED);
```

Comment:
why are we removing these lines?

Reply:
If you look at closely, we are not removing any lines here. git diff shows a little weird here.